### PR TITLE
Design task/task 2 updated

### PR DESCRIPTION
**Pull request by Conormc98**

### CODE REVIEW

File: ExpressAPI/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ var options = { cookies: true }
 var jwtGen = require('jsonwebtoken', options);
 var jwt = require('express-jwt');
 
-app.use(jwt({ secret: process.env.JWT_Secret}).unless({path: ['/login', '/SecureGenerateUser']}));
+app.use(jwt({ secret: process.env.JWT_Secret}).unless({path: ['/login', '/SecureGenerateUser', '/getBandRoles/4']}));
```
Comment (JamesMatchett) [inline ExpressAPI/index.js]:
```suggestion
app.use(jwt({ secret: process.env.JWT_Secret}).unless({path: ['/login', '/SecureGenerateUser']}));
```

**Review by JamesMatchett — CHANGES_REQUESTED** [review]

File: ExpressAPI/index.js (hunk 1 above)
Comment (JamesMatchett) [reply ExpressAPI/index.js]: Remove getBand from exclusion list

**Review by ChrisThompsonK — APPROVED** [review]
These changes look good to me 👍

**Review by JamesMatchett — APPROVED** [review]

File: ExpressAPI/index.js [hunk 2]
```diff
@@ -174,15 +174,23 @@ app.get('/InvalidateUserToken', function(req, res){
         db.clearUserToken(header);
         res.sendStatus(200)
     }
-})
+});
 
 app.get('/IsUserValid', function(req,res){
```
Comment (micalexis) [inline ExpressAPI/index.js:173]: refactor your endpoints to make them RESTful

File: kjourney/src/app/bands/bands.component.html [hunk 3]
```diff
@@ -1,68 +1,91 @@
 <router-outlet></router-outlet>
-<div class="topnav">
-  <a href="#">Your Role</a>
-  <a href="#">Support</a>
-</div>
+<header>
+  <br>
+  <br>
+  <h1 style="margin-top: 20px; text-align: center;" *ngFor="let band of bands">{{ band.bandName }}</h1>
```
Comment (micalexis) [inline kjourney/src/app/bands/bands.component.html]: why is this styling for the template not in the .css for the component?

File: kjourney/src/app/bands/bands.component.html [hunk 4]
```diff
@@ -1,68 +1,91 @@
 <router-outlet></router-outlet>
-<div class="topnav">
-  <a href="#">Your Role</a>
-  <a href="#">Support</a>
-</div>
+<header>
+  <br>
```
Comment (micalexis) [inline kjourney/src/app/bands/bands.component.html]: cant this be achieved with styling?

**Review by micalexis — CHANGES_REQUESTED** [review]
address comments

File: ExpressAPI/index.js (hunk 2 above)
Comment (micalexis) [reply ExpressAPI/index.js:173]: ?

File: kjourney/src/app/bands/bands.component.html [hunk 5]
```diff
@@ -1,68 +1,89 @@
...
       <tr>
-        <th>Band Level</th>
-        <th>Training Name</th>
-        <th>Training Type</th>
+        <th style="padding: 10px;">Training Name</th>
+        <th style="text-align: right">Training Type</th>
       </tr>
       </thead>
       <tbody>
       <tr class="myRow" id="pop-over" *ngFor="let trainingPerBands of trainingPerBand">
```
Comment (micalexis) [inline kjourney/src/app/bands/bands.component.html]: how can you get a plural from a singular?
change the names of the variables to make sense

File: kjourney/src/app/data.service.ts [hunk 6]
```diff
@@ -28,4 +30,12 @@ export class DataService {
   public getResponsibilities(bandID): Observable<Responsibility[]> {
     return this.http.get<Responsibility[]>('api/responsibilities/' + bandID);
   }
+
+  public getBandRoles(bandID): Observable<BandRole[]> {
+    return this.http.get<BandRole[]>('api/roles/' + bandID);
```
Comment (micalexis) [inline kjourney/src/app/data.service.ts]: you are getting roles by bandId so you need to use query parameters 

`api/roles/?bandId=`

**Review by micalexis — CHANGES_REQUESTED** [review]
address comments and squash commits

**Review by micalexis — APPROVED** [review]
squash commits

**Pull request merged**